generators/cloud-init.go

```diff
@@ -0,0 +1,108 @@
+package generators
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/lxc/lxd/shared/api"
+
+	"github.com/lxc/distrobuilder/image"
+	"github.com/lxc/distrobuilder/shared"
+)
+
+// CloudInitGenerator represents the cloud-init generator.
+type CloudInitGenerator struct{}
+
+// RunLXC does nothing.
+func (g CloudInitGenerator) RunLXC(cacheDir, sourceDir string, img *image.LXCImage,
+	defFile shared.DefinitionFile) error {
+	// no cloud-init support for LXC, ignoring generator
```

Comment:
We should actually figure out the right way to disable cloud-init in there as otherwise the container will just hang forever during boot trying to locate a data source.

Reply:
With OpenRC, we just remove any cloud-init related symlinks in `/etc/runlevels/*`. With systemd, we create `/etc/cloud/cloud-init.disabled` as described [here](https://cloudinit.readthedocs.io/en/latest/topics/boot.html#generator).